cmd/buildah/umount.go

```diff
@@ -2,52 +2,91 @@ package main
 
 import (
 	"fmt"
-	"os"
 
 	"github.com/pkg/errors"
-	"github.com/projectatomic/buildah/util"
 	"github.com/urfave/cli"
 )
 
 var (
+	umountFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "umount all of the currently mounted containers",
+		},
+	}
 	umountCommand = cli.Command{
 		Name:           "umount",
 		Aliases:        []string{"unmount"},
 		Usage:          "Unmounts the root file system on the specified working containers",
 		Description:    "Unmounts the root file system on the specified working containers",
 		Action:         umountCmd,
 		ArgsUsage:      "CONTAINER-NAME-OR-ID [...]",
+		Flags:          umountFlags,
 		SkipArgReorder: true,
 	}
 )
 
 func umountCmd(c *cli.Context) error {
+	umountAll := c.Bool("all")
 	umountContainerErrStr := "error unmounting container"
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("at least one container ID must be specified")
 	}
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass any container IDs")
+	}
 
 	store, err := getStore(c)
 	if err != nil {
 		return err
 	}
 
 	var lastError error
-	for _, name := range args {
+	if len(args) > 0 {
+		for _, name := range args {
+			builder, err := openBuilder(getContext(), store, name)
+			if err != nil {
+				if lastError != nil {
+					fmt.Println(lastError)
+				}
+				lastError = errors.Wrapf(err, "%s %s", umountContainerErrStr, name)
+				continue
+			}
+			if builder.MountPoint == "" {
+				continue
+			}
 
-		builder, err := openBuilder(getContext(), store, name)
+			id := builder.ContainerID
+			if err = builder.Unmount(); err != nil {
+				if lastError != nil {
+					fmt.Println(lastError)
+				}
+				lastError = errors.Wrapf(err, "%s %q", umountContainerErrStr, builder.Container)
+				continue
+			}
+			fmt.Printf("%s\n", id)
+		}
+	} else {
+		builders, err := openBuilders(store)
 		if err != nil {
-			lastError = util.WriteError(os.Stderr, errors.Wrapf(err, "%s %s", umountContainerErrStr, name), lastError)
-			continue
+			return errors.Wrapf(err, "error reading build Containers")
 		}
+		for _, builder := range builders {
+			if builder.MountPoint == "" {
+				continue
+			}
 
-		id := builder.ContainerID
-		if err = builder.Unmount(); err != nil {
-			lastError = util.WriteError(os.Stderr, errors.Wrapf(err, "%s %q", umountContainerErrStr, builder.Container), lastError)
-			continue
+			id := builder.ContainerID
+			if err = builder.Unmount(); err != nil {
+				if lastError != nil {
+					fmt.Println(lastError)
```

Comment:
These lines should still be 
fmt.Fprintln(os.Stderr, lastError)
Otherwise the error message goes to stdout, we want it at stderr.